### DISCUSSION TITLE
Timeline for PHREEQC integration?

### DISCUSSION THREAD

**Discussion by RyanConway91** [MODFLOW-ORG/modflow6#1352] [Q&A] [answered]
I forget when or where, but I recall someone from the MF6 dev team mentioning integrating PHREEQC capabilities into MF6 via the API. 

Can anyone provide an details on this timeline?

**Comment by christianlangevin** [comment, marked as answer]
This is something that we are starting to look into, but there is no real timeline for it.  There are a lot of little details that we will need to work through, and it's difficult to put a time estimate on them.

**Comment by FTW12345** [comment]
Hello,
Is the timeline for integrating PHREEQC and MF6 perhaps already taking shape?
I would like to simulate the geochemical processes that occur during density flow.
Until now, I ran into programs with limitations (PHAST, PHWAT, PHT3D).
Therefore, I hope I can use MF6+PHREEQC soon.
Thanks in advance,
MvdW

**Comment by wangzitao21** [comment]
I'm also looking forward to the integration of PHREEQC with MODFLOW 6.

**Comment by RyanConway91** [comment]
FYI, there is a MFUSG-PHREEQC integration called PHTUSG if you need reactive transport for an disv\disu type model

**Comment by JoerivanEngelen** [comment]
FYI, a little searching the internet for MODFLOW6 couplings with PHREEQC yielded some results (we are 2 years further of course).

Experimental & open-source from @wangzitao21:
[wangzitao21/mf6pqc: MF6PQC is a framework that couples MODFLOW6 with PHREEQCRM.](https://github.com/wangzitao21/mf6pqc)

Beta & open-source:
[rtmf6 - Reactive Transport Modeling with MODFLOW 6 and PHREEQC 3 — rtmf6 0.0.1 documentation](https://rtmf6.com/)

Beta & open-source:
[p-ortega/mf6rtm: A python package for reactive transport modeling via the MODFLOW 6 and PhreeqcRM APIs](https://github.com/p-ortega/mf6rtm)

FYI: I gave code that has been benchmarked and published this in the documentation a "Beta" status.

**Reply by FTW12345** [reply]
Thanks. Nice to read. I hope to apply the MODFLOW 6-PHREEQCRM combination soon to simulate hydrogeochemical reactions in a freshening aquifer, which must also include variable density flow.
Request: I suggest that the combination be integrated into iMOD Python to facilitate application.

**Reply by p-ortega** [reply]
Hey @FTW12345 and @JoerivanEngelen  

We welcome PRs and collaborators to https://github.com/p-ortega/mf6rtm, and I’m happy to help with using it. We’re releasing v0.2.0 soon, and we currently have six examples in the repo, with three more on the way—including a 3D study case using DISV and a NAPL study case.

**Reply by pya** [reply]
Good to see that there is some progress here. Small correction: rtmf6 is MIT licensed but not released yet, so technically not closed-source.  Disclaimer: I am the author of rtmf6. ;) @p-ortega maybe there is room for collaboration. I will have look at mf6rtm.

**Reply by JoerivanEngelen** [reply]
@FTW12345 nice to know that our package [iMOD Python](https://github.com/deltares/imod-python) is on your radar, but coupling kernels is outside its scope. We have the [iMOD Coupler](https://github.com/Deltares/imod_coupler) for that, but that mostly focuses on coupling MODFLOW6 to hydrological kernels, and not on geochemical ones. I think @p-ortega and his colleagues did a good job benchmarking coupled PHREEQC and MODFLOW6 simulations, so I would start seeing if their package fits (nearly) all your criteria.

@pya sorry for calling rtmf6 closed-source! From quick glance at the website, it wasn't really clear to me, I should have looked better! Changed the text from "closed-source" to "unreleased".

**Reply by p-ortega** [reply]
@pya Yeah, def! we are always open to collaborate.

**Reply by emorway-usgs** [reply]
Note there's a related issue over on the [modflow6api](https://github.com/MODFLOW-ORG/modflowapi) repo [here](https://github.com/MODFLOW-ORG/modflowapi/issues/86).  Though it looks like @JoerivanEngelen referenced @wangzitao21 repo earlier in this discussion, so I assume everyone is aware of Zitao's work as well.

**Reply by pya** [reply]
There is a new release of [rtmf6](https://docs.rtmf6.com/index.html). We added few new benchmarks, including [Uranium Release and Capture by Reduction](https://docs.rtmf6.com/benchmarks/uranium_release/index.html). More benchmarks are in preparation.

**Reply by JoerivanEngelen** [reply]
That's good to hear @pya, I updated the list!
